compiler/modules/nback/module.bal

```diff
@@ -102,10 +102,17 @@ function orderByFwdTargets(bir:BasicBlock[] blocks, bir:Label label, boolean[] v
     ordered.push(label);
 }
 
+isolated function index(UsedSemType ty) returns int {
+    return ty.index;
+}
+
 function createTypeUsage(table<UsedSemType> usedSemTypes) returns TypeUsage {
     byte[] uses = [];
     t:SemType[] types = [];
-    foreach var used in usedSemTypes {
+    // JBUG 36056 can't directly iterate over usedSemTypes
+    UsedSemType[] usedSemTy = usedSemTypes.toArray();
+    usedSemTy = usedSemTy.sort("ascending", index);
```

Comment:
I don't think you need this.  They are added to the table in order of increasing index. So the workaround can just change the next line to `usedSemTypes.toArray()`.

Reply:
I also thought the same but just for the failing testcase what I am getting from `usedSemTypes.toArray()` is not sorted.

Reply:
That should be reported as another bug (maybe the spec is not as clear as it should be here).